### ISSUE TITLE
##[error]RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded

### ISSUE THREAD

**Issue by logjeferson** [rfennell/AzurePipelines#1784]
### Azure DevOps Extensions

WIKI PDF Generator

### Platform

Azure DevOps Services

### Extension Version

4.0.3

### Describe the bug

When I run the export for a slightly larger number of wiki pages I receive the buffer limit error message:

System.ArgumentException: Markdown elements in the input are too deeply nested - depth limit exceeded. Input is most likely not sensible or is a very large table.

When I activate the debug I receive the log below:
##[error]RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded

### Repo Steps

...

**Comment by rfennell** [comment]
As I always ask, could you run your pipeline with a variable set `system.debug=true` or check the diagnostic box when queuing the run. Then add the detailed log (anonymized if needed) to this issue. This will show exactly when the fault occurs.

However, I in this case I suspect the issue will not be in my code. This is because my task wrappers {Max Melcher's AzureDevOps.WikiPDFExport CLI tool](https://github.com/MaxMelcher/AzureDevOps.WikiPDFExport) to do the actual export. So you will need to log an issue in his repo.

I have checked the Max's repo and see there is already a similar issue raised MaxMelcher/AzureDevOps.WikiPDFExport#127,but unfortunately no issue.

**Comment by logjeferson** [comment]
Right,
Below are the debug logs:

##[debug]Evaluating condition for step: 'Export Wiki Files to PDF'
##[debug]Evaluating: SucceededNode()
##[debug]Evaluating SucceededNode:
##[debug]=> True
##[debug]Result: True
Starting: Export Wiki Files to PDF
==============================================================================
Task         : A wrapper for the AzureDevOps.WikiPDFExport
Description  : A wrapper for the [AzureDevOps.WikiPDFExport](https://github.com/MaxMelcher/AzureDevOps.WikiPDFExport)
Version      : 3.8.5
Author       : Richard Fennell
Help         : Version: 3.8.5. [More Information](https://github.com/rfennell/vNextBuild/wiki/Wiki-PdfExport-Task/)
==============================================================================
##[debug]Using node path: C:\agents\3.236.1\externals\node16\bin\node.exe
##[debug]agent.TempDirectory=D:\a\_temp
##[debug]loading inputs and endpoints
##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
##[debug]loading INPUT_CLONEREPO
##[debug]loading INPUT_DOWNLOADPATH
##[debug]loading INPUT_EXTRAPARAMETERS
##[debug]loading INPUT_INJECTEXTRAHEADER
##[debug]loading INPUT_LOCALPATH
##[debug]loading INPUT_OUTPUTFILE
##[debug]loading INPUT_USEAGENTTOKEN
##[debug]loading INPUT_USEPRERELEASE
##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
##[debug]loading SECRET_UIPATHCLIENTID
##[debug]loading SECRET_UIPATHSECRETID
##[debug]loaded 14
##[debug]Agent.ProxyUrl=undefined
##[debug]Agent.CAInfo=undefined
##[debug]Agent.ClientCert=undefined
##[debug]Agent.SkipCertValidation=undefined
##[debug]repo=undefined
##[debug]localpath=D:\a\1\s\Internal-Documentation
##[debug]singleFile=undefined
##[debug]outputFile=D:\a\1\s\321067.pdf
##[debug]extraParameters=--verbose
##[debug]user=undefined
##[debug]password=undefined
##[debug]useAgentToken=true
##[debug]branch=undefined
##[debug]injectExtraHeader=False
##[debug]cloneRepo=false
##[debug]overrideExePath=undefined
##[debug]downloadPath=D:\a\_temp
##[debug]rootExportPath=undefined
##[debug]usePreRelease=false
Defaulting variable rootExportPath as not defined
Variable: Repo [undefined]
Variable: Use Agent Token [true]
Variable: Username [undefined]
Variable: Password [*****]
Variable: LocalPath [D:\a\1\s\Internal-Documentation]
Variable: rootExportPath [D:\a\1\s\Internal-Documentation]
Variable: SingleFile [undefined]
Variable: OutputFile [D:\a\1\s\321067.pdf]
Variable: Branch [undefined]
Variable: InjectExtraHeader [false]
Variable: OverrideExePath [undefined]
Variable: UsePreRelease [false]
Variable: DownloadPath [D:\a\_temp]
##[debug]AGENT.OS=Windows_NT
Starting download azuredevops-export-wiki.exe to D:\a\_temp
Downloading MaxMelcher/AzureDevOps.WikiPDFExport@4.6.0...
Download done
##[debug]Agent environment resources - Disk: D:\ Available 11998.00 MB out of 14333.00 MB, Memory: Used 1923.00 MB out of 7167.00 MB, CPU: Usage 15.50%
##[debug]Agent environment resources - Disk: D:\ Available 11998.00 MB out of 14333.00 MB, Memory: Used 1839.00 MB out of 7167.00 MB, CPU: Usage 13.06%
Downloaded executable D:\a\_temp\azuredevops-export-wiki.exe
Using the EXE path of 'D:\a\_temp\azuredevops-export-wiki.exe' for AzureDevOps.WikiPDFExport
Processing the contents of the folder 'D:\a\1\s\Internal-Documentation'
Using D:\a\1\s\Internal-Documentation as the root path to export
Using the .order file in the root of the wiki to determine the order of the pages to export
Adding extra parameters to the command line
Changing folder to D:\a\1\s\Internal-Documentation
Using command '"D:\a\_temp\azuredevops-export-wiki.exe"  -p "D:\a\1\s\Internal-Documentation" -o "D:\a\1\s\321067.pdf" --verbose'
##[debug]Agent environment resources - Disk: D:\ Available 11998.00 MB out of 14333.00 MB, Memory: Used 1777.00 MB out of 7167.00 MB, CPU: Usage 11.28%
##[debug]Agent environment resources - Disk: D:\ Available 11998.00 MB out of 14333.00 MB, Memory: Used 1744.00 MB out of 7167.00 MB, CPU: Usage 9.93%
##[debug]Agent environment resources - Disk: D:\ Available 11998.00 MB out of 14333.00 MB, Memory: Used 1783.00 MB out of 7167.00 MB, CPU: Usage 8.87%
00:00:02.41 ERR: Something bad happend.
System.ArgumentException: Markdown elements in the input are too deeply nested - depth limit exceeded. Input is most likely not sensible or is a very large table.
   at Markdig.Helpers.ThrowHelper.<CheckDepthLimit>g__DepthLimitExceeded|16_0()
   at Markdig.Renderers.RendererBase.WriteChildren(ContainerInline containerInline)
   at Markdig.Renderers.Html.Inlines.DelimiterInlineRenderer.Write(HtmlRenderer renderer, DelimiterInline obj)
   at Markdig.Renderers.MarkdownObjectRenderer`2.Write(RendererBase renderer, MarkdownObject obj)
   at Markdig.Renderers.RendererBase.Write(MarkdownObject obj)
   at Markdig.Renderers.RendererBase.WriteChildren(ContainerInline containerInline)
   at Markdig.Renderers.RendererBase.Write(MarkdownObject obj)
   at Markdig.Renderers.Html.ParagraphRenderer.Write(HtmlRenderer renderer, ParagraphBlock obj)
   at Markdig.Renderers.MarkdownObjectRenderer`2.Write(RendererBase renderer, MarkdownObject obj)
   at Markdig.Renderers.RendererBase.Write(MarkdownObject obj)
   at Markdig.Renderers.RendererBase.WriteChildren(ContainerBlock containerBlock)
   at Markdig.Renderers.RendererBase.Write(MarkdownObject obj)
   at Markdig.Renderers.TextRendererBase.Render(MarkdownObject markdownObject)
   at azuredevops_export_wiki.MarkdownConverter.ConvertToHTML(IList`1 files)
   at azuredevops_export_wiki.WikiPDFExporter.Export()

##[error]Error: Command failed: "D:\a\_temp\azuredevops-export-wiki.exe"  -p "D:\a\1\s\Internal-Documentation" -o "D:\a\1\s\321067.pdf" --verbose
##[debug]Processed: ##vso[task.issue type=error;source=TaskInternal;]Error: Command failed: "D:\a\_temp\azuredevops-export-wiki.exe"  -p "D:\a\1\s\Internal-Documentation" -o "D:\a\1\s\321067.pdf" --verbose

##[debug]task result: Failed
##[error]Error: Command failed: "D:\a\_temp\azuredevops-export-wiki.exe"  -p "D:\a\1\s\Internal-Documentation" -o "D:\a\1\s\321067.pdf" --verbose

##[debug]Processed: ##vso[task.issue type=error;source=TaskInternal;]Error: Command failed: "D:\a\_temp\azuredevops-export-wiki.exe"  -p "D:\a\1\s\Internal-Documentation" -o "D:\a\1\s\321067.pdf" --verbose
##[debug]Processed: ##vso[task.complete result=Failed;]Error: Command failed: "D:\a\_temp\azuredevops-export-wiki.exe"  -p "D:\a\1\s\Internal-Documentation" -o "D:\a\1\s\321067.pdf" --verbose

**Comment by rfennell** [comment]
Yes, as suspected the error is coming from the execution of the underlying tool

```
Using command '"D:\a_temp\azuredevops-export-wiki.exe" -p "D:\a\1\s\Internal-Documentation" -o "D:\a\1\s\321067.pdf" --verbose'
```
There is nothing I can do other than suggest logging an issue on https://github.com/MaxMelcher/AzureDevOps.WikiPDFExport

Remember there is nothing stopping you downloading the CLI and running it on your PC to experiment without the need to run the pipeline
